spark/v3.5/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestFastForwardBranchProcedure.java

```diff
@@ -247,4 +234,39 @@ public void testFastForwardNonMain() {
     assertThat(outputRow.get(1)).isEqualTo(branch1Snapshot.snapshotId());
     assertThat(outputRow.get(2)).isEqualTo(branch2Snapshot.snapshotId());
   }
+
+  @Test
+  public void testFastForwardNonExistingFromMainCreatesBranch() {
+    sql("CREATE TABLE %s (id int NOT NULL, data string) USING iceberg", tableName);
+    String branch1 = "branch1";
+    sql("ALTER TABLE %s CREATE BRANCH %s", tableName, branch1);
+    String branchIdentifier = String.format("%s.branch_%s", tableName, branch1);
+    sql("INSERT INTO TABLE %s VALUES (1, 'a')", branchIdentifier);
+    sql("INSERT INTO TABLE %s VALUES (2, 'b')", branchIdentifier);
+    Table table = validationCatalog.loadTable(tableIdent);
+    table.refresh();
+    Snapshot branch1Snapshot = table.snapshot(branch1);
+
+    List<Object[]> output =
+        sql(
+            "CALL %s.system.fast_forward('%s', '%s', '%s')",
+            catalogName, tableIdent, SnapshotRef.MAIN_BRANCH, branch1);
+    List<Object> outputRow = Arrays.stream(output.get(0)).collect(Collectors.toList());
```

Comment:
can this be simplified to `assertThat(sql(...)).containsExactly(row(SnapshotRef.MAIN_BRANCH, null, branch1Snapshot.snapshotId()));`?

Reply:
It sure can, good suggestion, Updated!